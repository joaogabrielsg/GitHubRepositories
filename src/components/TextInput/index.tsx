import React from 'react';
import {
  TextInput,
  TextInputProps,
  StyleSheet,
  View,
  Image,
  TouchableWithoutFeedback,
} from 'react-native';

import Search from '../../../assets/search.png';

import {theme, font} from '../../styles';

interface CustomTextInputProps extends TextInputProps {
  text: string;
  onChangeText?: () => void;
  onSearch?: () => void;
}

const CustomTextInput: React.FunctionComponent<CustomTextInputProps> = props => {
  return (
    <View style={styles.container}>
      <TextInput
        style={styles.textInput}
        value={props.text}
        onChangeText={props.onChangeText}
        testID="nativeTextInput"
      />
      <TouchableWithoutFeedback onPress={props.onSearch}>
        <Image style={styles.image} source={Search} />
      </TouchableWithoutFeedback>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 20,
    backgroundColor: theme.secondary,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  textInput: {
    flex: 1,
    fontSize: font.subTitle,
    color: theme.textColor,
  },
  image: {
    height: 30,
    width: 30,
    resizeMode: 'contain',
  },
});

export default CustomTextInput;
