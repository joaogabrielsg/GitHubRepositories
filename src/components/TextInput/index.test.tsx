import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';

import TextInput from './index';

describe('TextInput component', () => {
  const onChangeTextMock = jest.fn();

  it('to match snapshot', () => {
    const {toJSON} = render(
      <TextInput text="" onChangeText={onChangeTextMock} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('to text prop is pass to TextInput value prop', async () => {
    const {getByTestId} = render(
      <TextInput text="text test" onChangeText={onChangeTextMock} />,
    );

    const nativeTextInput = getByTestId('nativeTextInput');
    expect(nativeTextInput.props.value).toBe('text test');
  });

  it('to onChangeText be called on change text', async () => {
    const {getByTestId} = render(
      <TextInput text="text test" onChangeText={onChangeTextMock} />,
    );

    const nativeTextInput = getByTestId('nativeTextInput');
    fireEvent.changeText(nativeTextInput, '');

    expect(onChangeTextMock).toHaveBeenCalled();
  });
});
