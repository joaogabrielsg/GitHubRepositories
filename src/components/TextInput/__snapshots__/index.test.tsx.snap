// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component to match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(255, 255, 255)",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 20,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[MockFunction]}
    rejectResponderTermination={true}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="nativeTextInput"
    underlineColorAndroid="transparent"
    value=""
  />
  <Image
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../assets/search.png",
      }
    }
    style={
      Object {
        "height": 30,
        "resizeMode": "contain",
        "width": 30,
      }
    }
  />
</View>
`;
